imap/jmap_api.c

```diff
@@ -1080,25 +1056,7 @@ void jmap_add_id(jmap_req_t *req, const char *creation_id, const char *id)
 HIDDEN int jmap_openmbox(jmap_req_t *req, const char *name,
                          struct mailbox **mboxp, int rw)
 {
-    int i, r;
-    struct _mboxcache_rec *rec;
-
-    for (i = 0; i < req->mboxes->count; i++) {
-        rec = (struct _mboxcache_rec*) ptrarray_nth(req->mboxes, i);
-        if (!strcmp(name, mailbox_name(rec->mbox))) {
-            if (rw && !rec->rw) {
-                /* Lock promotions are not supported */
-                syslog(LOG_ERR, "jmapmbox: failed to grab write-lock"
-                       " on cached read-only mailbox %s", name);
-                return IMAP_INTERNAL;
-            }
-            /* Found a cached mailbox. Increment refcount. */
-            rec->refcount++;
-            *mboxp = rec->mbox;
-
-            return 0;
-        }
-    }
+    int r;
 
     /* Add mailbox to cache */
     if (req->force_openmbox_rw)
```

Comment:
this conditional and the `rw` variable is obsolete, this could just become

     r = req->force_openmbox_rw ? mailbox_open_iwl(name, mboxp) : mailbox_open_irl(name, mboxp);

Reply:
OK?  do we ALWAYS set force_openmbox_rw?  It looks like `rw` gets set to both possible values a lot

```
8109:    int r = jmap_openmbox_by_guidrec(rock->req, rec, &mbox, /*rw*/0);
8151:            r = jmap_openmbox(req, mboxname, &mbox, 0);
8838:    r = jmap_openmbox(req, mboxname, &mbox, 1);
8982:        r = jmap_openmbox(req, dstname, &dst, 1);
12402:            if (!r) r = jmap_openmbox(req, mboxrec->mboxname, &mbox, /*rw*/1);
12477:                if (!r) jmap_openmbox(req, mbentry->name, &mbox, /*rw*/1);
13087:        int r = jmap_openmbox(req, mboxrec->mboxname, &mbox, 1);
13611:    int r = jmap_openmbox_by_guidrec(rock->req, rec, &mbox, /*rw*/1);
13742:        r = jmap_openmbox(req, mbentry->name, &mbox, 0);
13791:    r = jmap_openmbox(req, mbentry->name, &mbox, 0);
13966:            r = jmap_openmbox(req, dst_mboxname, &dst_mbox, /*rw*/1);
```

Reply:
oops, my bad. sorry, I was wrong. so this would get ` = (rw || req->force_openmbox_rw) ? ..` then?

Reply:
Or lets get rid of `force_openmbox_rw` altogether which would also allow to remove `jmap_openmbox`